ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,511 @@
+    name text NOT NULL,
+    cost_model text NOT NULL,
+    state text NOT NULL,
+    monthly_budget bigint,
+    blacklisted_site_urls text[],
+    created_at timestamp without time zone NOT NULL,
+    updated_at timestamp without time zone NOT NULL
+  );
+
+  CREATE TABLE ads (
+    id bigserial PRIMARY KEY,
+    campaign_id bigint REFERENCES campaigns (id),
+    name text NOT NULL,
+    image_url text,
+    target_url text,
+    impressions_count bigint DEFAULT 0,
+    clicks_count bigint DEFAULT 0,
+    created_at timestamp without time zone NOT NULL,
+    updated_at timestamp without time zone NOT NULL
+  );
+
+  CREATE TABLE clicks (
+    id bigserial PRIMARY KEY,
+    ad_id bigint REFERENCES ads (id),
+    clicked_at timestamp without time zone NOT NULL,
+    site_url text NOT NULL,
+    cost_per_click_usd numeric(20,10),
+    user_ip inet NOT NULL,
+    user_data jsonb NOT NULL
+  );
+
+  CREATE TABLE impressions (
+    id bigserial PRIMARY KEY,
+    ad_id bigint REFERENCES ads (id),
+    seen_at timestamp without time zone NOT NULL,
+    site_url text NOT NULL,
+    cost_per_impression_usd numeric(20,10),
+    user_ip inet NOT NULL,
+    user_data jsonb NOT NULL
+  );
+
+There are modifications we can make to the schema which will give it a performance boost in a distributed environment like Citus. To see how, we must become familiar with how Citus distributes data and executes queries.
+
+Scaling the Relational Data Model
+---------------------------------
+
+The relational data model is great for applications. It protects data integrity, allows flexible queries, and accommodates changing data. Traditionally the only problem was that relational databases weren't considered capable of scaling to the workloads needed for big SaaS applications. Developers had to put up with awkward NoSQL databases -- or a collection of backend services -- to reach that size. With Citus you can keep your data model *and* make it scale.
+
+Citus appears to applications as a single PostgreSQL database, but it internally routes queries to an adjustible number of physical servers (nodes) which can process requests in parallel. When configuring Citus, the database administrator specifies how the routing will happen.
+
+Multi-tenant applications have a nice property that we can take advantage of: queries usually always request information for one tenant at a time, not a mix of tenants. For instance, when a visitor is shopping at an online store runing on an ecommerce platform, the search results are just for that store; competitors' products are not mixed in.
+
+Because application queries are restricted to a single tenant, such as a store or company, one approach for making multi-tenant application queries fast is to store *all* data for a given tenant on the same node. This minimizes network overhead between the nodes. The joins, key constraints, and transactions can happen entirely within a node and do not require information from other nodes.
```

Comment:
Yeah it is a big section of raw text, but the existing diagrams use a different example than our ad analytics so it might be kind of confusing to drop them in there verbatim.